### PR TITLE
fix(v2): responsive search bar

### PR DESCRIPTION
**Pull request by shirshendubhowmick**
## Motivation

This PR tries to solve isssue #1697 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

For devices with viewport width less than 360px, the searchbar is replaced with a search icon.
On tap of the search icon the searchbar is expanded and the text "Docusaurus" beside the logo is hidden and remains hidden while the search bar is expanded.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)

### CODE REVIEW

File: packages/docusaurus-theme-search-algolia/src/theme/SearchBar/styles.css [hunk 1]
```diff
@@ -44,6 +44,41 @@
...
+  background-position: center;
+  display: none;
+}
+
+.search-icon-hidden {
+  visibility: hidden;
+}
+
+@media (max-width: 360px) {
+
```
Comment (endiliey) [inline packages/docusaurus-theme-search-algolia/src/theme/SearchBar/styles.css]:
```suggestion
```

**Comment by shirshendubhowmick** [comment]
@endiliey Sure, will look into it. Let me know if we can improve more transition / animation.

**Comment by shirshendubhowmick** [comment]
@endiliey Can work on refactoring with Hooks. Let me know when we have an issue created for this.

**Comment by endiliey** [comment]
Just send a PR. an issue isnt always necessary :p

**Comment by shirshendubhowmick** [comment]
Sure, let me know what are the components that we planning to refactor

**Comment by endiliey** [comment]
https://github.com/facebook/docusaurus/blob/master/packages/docusaurus-theme-search-algolia/src/theme/SearchBar/index.js

**Pull request merged**